assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -0,0 +1,312 @@
+  const recentUsedFonts = useMemo(() => {
+    const fontsFromSlug = recentUsedFontValues.map((value) =>
+      fonts.find((font) => font.value === value)
+    );
+    return searchValue && searchValue !== ''
+      ? fontsFromSlug.filter(({ name }) =>
+          name.toLowerCase().includes(searchValue.toLowerCase())
+        )
+      : fontsFromSlug;
+  }, [searchValue, recentUsedFontValues, fonts]);
+
+  // Font options that start with search term
+  const otherFonts = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? fonts.filter(({ name }) =>
+            name.toLowerCase().startsWith(searchValue.toLowerCase())
+          )
+        : fonts,
+    [searchValue, fonts]
+  );
+
+  // Font options that include search term but not start with
+  const includeSearchFonts = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? fonts.filter(
+            ({ name }) =>
+              name.toLowerCase().indexOf(searchValue.toLowerCase()) > 0
+          )
+        : [],
+    [searchValue, fonts]
+  );
+
+  useEffect(() => {
+    const combinedFontList = [
+      ...recentUsedFonts,
+      ...otherFonts,
+      ...includeSearchFonts,
+    ]
+      .slice(currentIndex, currentIndex + 10)
+      .filter(({ service }) => service.includes('google'))
+      .map(({ name }) => name);
+
+    if (combinedFontList.length > 0) {
+      getMenuFonts(combinedFontList).then(setMenuFonts);
+    }
+  }, [
+    currentIndex,
+    getMenuFonts,
+    recentUsedFonts,
+    otherFonts,
+    includeSearchFonts,
+  ]);
+
+  const handleScroll = useCallback(() => {
+    const listElement = listContainerRef.current;
+    let { scrollTop } = listElement;
+    // Remove the Ul list container padding offset
+    scrollTop -= LIST_PADDING;
+    // Calculate the current visible font index
+    let currentVisibleIndex = Math.floor(scrollTop / FONT_ROW_HEIGHT);
+
+    // If there is recent used fonts and current visible options index is on next list
+    if (
+      recentUsedFonts.length > 0 &&
+      currentVisibleIndex > recentUsedFonts.length
+    ) {
+      scrollTop -= LIST_PADDING * 2;
+      currentVisibleIndex =
+        Math.floor(scrollTop / FONT_ROW_HEIGHT) - recentUsedFonts.length;
+    }
+    // If there is other fonts and current visible options index is on next list
+    if (otherFonts.length > 0 && currentVisibleIndex > otherFonts.length) {
+      scrollTop -= LIST_PADDING * 2;
+    }
+    currentVisibleIndex = Math.floor(scrollTop / FONT_ROW_HEIGHT);
+    if (currentVisibleIndex < 0) {
+      currentVisibleIndex = 0;
+    }
+    if (currentVisibleIndex !== currentActiveRef.current) {
+      currentActiveRef.current = currentVisibleIndex;
+      setCurrentIndex(currentVisibleIndex);
+    }
+  }, [otherFonts, recentUsedFonts]);
+
+  useEffect(() => {
+    inputRef.current.focus();
+    const listElement = listContainerRef.current;
+    listElement.addEventListener('scroll', handleScroll, { passive: false });
```

Comment:
You'd either have to remove `{passive: false}` here. Or add it to the `removeEventListener` as well.